core/domain/translation_domain.py

```diff
@@ -18,9 +18,280 @@
 
+import enum
+
+from core import feconf
 from core import utils
 
-from typing import Dict
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+class TranslatableContentFormat(enum.Enum):
+    """Represents all possible data types for any translatable content."""
+
+    HTML = 'html'
+    UNICODE_STRING = 'unicode'
+    SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content_value: feconf.ContentValueType
+    content_type: TranslatableContentFormat
+
+
+class TranslatableContent:
+    """TranslatableContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding translatable content value.
+        content_value: ContentValueType. The content value which can be
+            translated.
+        content_type: TranslatableContentFormat. The type of the
+            corresponding content value.
+    """
+
+    def __init__(
+        self,
+        content_id: str,
+        content_value: feconf.ContentValueType,
+        content_type: TranslatableContentFormat
+    ) -> None:
+        self.content_id = content_id
+        self.content_value = content_value
+        self.content_type = content_type
+
+    @classmethod
+    def from_dict(
+        cls,
+        translatable_content_dict: TranslatableContentDict
+    ) -> TranslatableContent:
+        """Returns a TranslatableContent object from its dict representation.
+
+        Args:
+            translatable_content_dict: dict. Dict representation of
+                TranslatableContent object.
+
+        Returns:
+            TranslatableContent. An instance of TranslatableContent class.
+        """
+        return cls(
+            translatable_content_dict['content_id'],
+            translatable_content_dict['content_value'],
+            translatable_content_dict['content_type'])
+
+    def to_dict(self) -> TranslatableContentDict:
+        """Returns the dict representation of TranslatableContent object.
+
+        Returns:
+            TranslatableContentDict. A dict, mapping content_id, content_value
+            and content_type of a TranslatableContent instance to
+            corresponding keys 'content_id', 'content_value' and
+            'content_type'.
+        """
+        return {
+            'content_id': self.content_id,
+            'content_value': self.content_value,
+            'content_type': self.content_type
+        }
+
+
+class TranslatedContent:
+    """Represents the translated content of the TranslatableContent object.
+
+    Args:
+        content_value: ContentValueType. Represents already-translated
+            content in any BaseTranslatableObject.
+        needs_update: bool. A boolean value represents whether any translation
```

Comment:
You don't need to repeat "a boolean value", just start the description with "Whether".

Also, just realized something that made me a bit confused. We don't store the language of the translation here? If a piece of content is translated in language A but not language B then does it count as "translated content" or not?

Reply:
Updated.

Yes, it will be counted as translated content. Basically translated content means content as a result of translation.
For example: 
I am a boy. (English) --> Soy un chico.(Spanish)  [TranslatedContent]